bittide/src/Bittide/SwitchDemoProcessingElement.hs

```diff
@@ -0,0 +1,128 @@
+-- SPDX-FileCopyrightText: 2025 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+{-# LANGUAGE NumericUnderscores #-}
+
+module Bittide.SwitchDemoProcessingElement where
+
+import Clash.Prelude
+
+import Data.Maybe (fromMaybe)
+
+-- | Simple processing element used for the Bittide switch demo.
+switchDemoPe ::
+  forall nodes dom.
+  ( HiddenClockResetEnable dom
+  , 1 <= nodes
+  ) =>
+  SNat nodes ->
+  -- | Local clock cycle counter
+  Signal dom (Unsigned 64) ->
+  -- | Incomming crossbar link
+  Signal dom (BitVector 64) ->
+  -- | Device DNA
+  Signal dom (Maybe (BitVector 96)) ->
+  -- | When to read from the crossbar link
+  Signal dom (Unsigned 64) ->
+  -- | How many cycles to read from crossbar link
+  Signal dom (Index (nodes * 3)) ->
+  -- | When to write to the crossbar link
+  Signal dom (Unsigned 64) ->
+  -- | How many cycles to read from crossbar link
```

Comment:
```suggestion
  -- | How many tri-cycles to write to crossbar link. Includes writing \"own\" data.
```

So three changes:

1. cycles -> tri-cycles (as discussed on Slack, including it here for completeness sake)
2. read -> write
3. +clarification on whether the cycles include writing own data or not